vscode/test/suites/language-service/notebook.test.ts

```diff
@@ -116,4 +117,46 @@ suite("Q# Notebook Tests", function suite() {
     assert.equal(location.range.start.line, 2);
     assert.equal(location.range.start.character, 10);
   });
+
+  test("Notebook uses Unrestricted profile by default even when workspace is Base", async () => {
+    // Store the original workspace target profile to restore it later
+    const originalConfig = vscode.workspace.getConfiguration("Q#");
+    const originalTarget = originalConfig.get("qir.targetProfile");
+    
+    try {
+      // Set workspace to base profile (restrictive)
+      await setTarget("base");
```

Comment:
Weird mixing and matching of patterns here - you're using setTarget from the extension code, but using the VS Code api to _get_ the same setting. Just go against the vscode api for both. Avoid reaching into extension internals in these integration tests.